### ISSUE TITLE
Unpredictable imports behaviour when importing file: depends on base image

### ISSUE THREAD

**Issue by distorhead** [flant/werf#2192]
Given following config:

```
import:
- artifact: myartifact
  add: /path/to/myfile
  to: /new/path
```

If `/new/path` is existing directory in the base image, then `/new/path/myfile` will be created.
If `/new/path` does not exist in the base image, then `/new/path` file will be created with the content of `/path/to/myfile`.

Expected more declarative behaviour: always create `/new/path` with the content of the `/path/to/myfile`.  In the case when `/new/path` is existing directory werf should exit with error.

**Comment by alexey-igrychev** [comment]
Answered and fixed in docs: #7710.

The behaviour isn't actually unpredictable, it just wasn't written down — and it differs by source type:

- a **directory** source has its contents merged into `to`, which is deterministic (werf appends a trailing slash to the rsync source spec when it detects a directory, `pkg/build/import_server/rsync_server.go`); this is also what allows several imports into the same target directory;
- a **file** source is what you hit here: werf creates only the parent of `to`, so if the base image happens to have `to` as a directory the file lands inside it, otherwise the file is created at exactly `to`. That mirrors Dockerfile `COPY` and is intentional.

v3 additionally turned the neighbouring ambiguity into a hard error: a trailing slash in `to:` now fails the build instead of being silently reinterpreted.

All three rules are now documented on the imports reference page in both locales. This issue will close once #7710 is merged — thanks for the original report.
